File: solutions/ruby/raindrops/1/raindrops.rb
module Raindrops

  def self.convert number
    sounds = {
      3 => "Pling",
      5 => "Plang",
      7 => "Plong",
    }
    raindrops = sounds.map{ |n, sound| sound if divisible(number, n)}.join 
    raindrops.empty? ? number.to_s : raindrops
  end

  def self.divisible(a, b)
    a % b == 0
  end

end

File: solutions/python/raindrops/1/raindrops.py
def convert(number):
    sounds = {
        3: "Pling",
        5: "Plang",
        7: "Plong",
    }
    raindrop_sound = str()
    for n, sound in sounds.items():
        if divisible(number, n):
            raindrop_sound += sound
    return raindrop_sound if raindrop_sound else str(number)

def divisible(a: int, b: int) -> bool:
    return True if not a % b else False
